Modules/mmapmodule.c

```diff
@@ -1562,12 +1562,14 @@ new_mmap_object(PyTypeObject *type, PyObject *args, PyObject *kwdict)
     off_lo = (DWORD)(offset & 0xFFFFFFFF);
     /* For files, it would be sufficient to pass 0 as size.
        For anonymous maps, we have to pass the size explicitly. */
+    Py_BEGIN_ALLOW_THREADS
     m_obj->map_handle = CreateFileMappingA(m_obj->file_handle,
                                            NULL,
                                            flProtect,
                                            size_hi,
                                            size_lo,
                                            m_obj->tagname);
+    Py_END_ALLOW_THREADS
```

Comment:
This should be after the `MapViewOfFile` call. We can probably restructure the function to store results in locals and then determine success/failure after we have the GIL back